linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -115,6 +115,22 @@ void Process ()
 				ProcessQueue ();
 				ProcessVirtualMethods ();
 			}
+
+			// deal with [TypeForwardedTo] pseudo-attributes
+			foreach (AssemblyDefinition assembly in _context.GetAssemblies ()) {
+				if (!assembly.MainModule.HasExportedTypes)
+					continue;
+
+				foreach (var exported in assembly.MainModule.ExportedTypes) {
+					if (!exported.IsForwarder)
+						continue;
+					var type = exported.Resolve ();
+					if (!Annotations.IsMarked (type))
+						continue;
+					Annotations.Mark (exported);
+					Annotations.Mark (assembly.MainModule);
```

Comment:
Thank you for the explanation. I run into similar issue and wasn't sure about the reasoning/history there.

I noticed it was even there in the original MonoTouchMarkStep.cs (https://github.com/xamarin/monotouch/commit/49fc552e1d43bc1dd39b3e12552d16cb9ac7dc53), but I am not sure why it was removed later (there were many moves of this code so it is not easy to follow)